python-sdk/src/langwatch/telemetry/context.py

```
@@ -50,10 +62,78 @@ def get_current_span() -> "LangWatchSpan":
+        and trace.trace_id
+        and trace.trace_id in main_thread_langwatch_span
+        and not _is_on_child_thread()
+    ):
+        main_thread_langwatch_span[trace.trace_id].pop()
+
+    try:
+        stored_langwatch_span.reset(token)
+    except Exception as e:
+        # Only warn if it's not a context error
+        if "different Context" not in str(e):
+            warnings.warn(f"Failed to reset LangWatch span context: {e}")
```

Ruff (0.11.9)

No explicit `stacklevel` keyword argument found

Set `stacklevel=2`

(B028)